### PR TITLE
[cmds][scd/store] Add 'db-manager scd-evict' subcommand enabling listing and deletion of expired SCD entities

### PR DESCRIPTION
**Pull request by mickmis**
Addresses #1074 

This PR adds a CLI tool `db-manager scd-evict` that enables deletion of expired entities. See enclosed README for more details on its usage.

Do note the approach taken when entities do not have end times: the last update times are used.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,8 +1,8 @@
 # This Dockerfile builds the InterUSS `dss` image which contains the binary
-# executables for the core-service and the db-manager. It also
+# executables for the core-service, the db-manager and the db-evictor. It also
```
[the remark this answers is not in the document]
Comment (mickmis) [reply Dockerfile:2]: Please see #1122. I will rework this PR once #1122 is accepted and merged.

File: cmds/db-evictor/main.go [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+	dssmodels "github.com/interuss/dss/pkg/models"
+	scdmodels "github.com/interuss/dss/pkg/scd/models"
+	"github.com/interuss/dss/pkg/scd/repos"
+	scdc "github.com/interuss/dss/pkg/scd/store/cockroach"
+)
+
+var (
+	listOpIntents = flag.Bool("op_intents", true, "set this flag to true to list expired operational intents")
+	listScdSubs   = flag.Bool("scd_subs", true, "set this flag to true to list expired SCD subscriptions")
+	ttl           = flag.Duration("ttl", time.Hour*24*112, "time-to-live duration used for determining expiration")
```
[the remark this answers is not in the document]
Comment (mickmis) [reply cmds/db-evictor/main.go:21]: That's based on @BenjaminPelletier's comment: https://github.com/interuss/dss/issues/1074#issuecomment-2305399224
This is basically a loose upper bound of 'no DSS deployment should ever need to have entities older than X'.
I've added a note about the default value, but I'd say that more details on determining that value should rather go into some operations documentation.

**Comment by mickmis** [comment]
FYI still need to test manually the CLI + ass unit tests.

**Comment by callumdmay** [comment]
Hey @mickmis where are we currently at with this tool? We just updated to v0.18 but we've accumulated a significant number of dangling subscriptions over the last month and they are having a significant affect on latency so we're hoping to leverage this tool ASAP

**Comment by mickmis** [comment]
Hi @callumdmay, sorry I've had to slow down significantly my work on this issue for the last few weeks. I've actually resumed that last Thursday and I expect this PR to be merged by tomorrow.

**Comment by mickmis** [comment]
@barroco this is now ready for another review. I've added unit tests and tested the tool manually in addition.

File: cmds/db-manager/cleanup/scd-evict.go [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+	return nil
+}
+
+func getSCDStore(ctx context.Context) (*scdc.Store, error) {
+	connectParameters := crdbflags.ConnectParameters()
+	connectParameters.ApplicationName = "db-manager"
+	connectParameters.DBName = scdc.DatabaseName
+	scdCrdb, err := cockroach.Dial(ctx, connectParameters)
+	if err != nil {
+		return nil, fmt.Errorf("failed to connect to database with %+v: %w", connectParameters, err)
```
[the remark this answers is not in the document]
Comment (mickmis) [reply cmds/db-manager/cleanup/scd-evict.go:106]: Good catch thanks. I propose to just not log it. Wrong DB credentials will fail fast and checking what credentials are passed exactly should not be too hard.

File: cmds/db-manager/cleanup/scd-evict.go [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+		expiredSubs      []*scdmodels.Subscription
+	)
+	action := func(ctx context.Context, r repos.Repository) (err error) {
+		if *listOpIntents {
+			expiredOpIntents, err = r.ListExpiredOperationalIntents(ctx, threshold)
+			if err != nil {
+				return fmt.Errorf("listing expired operational intents: %w", err)
+			}
+			if *deleteExpired {
+				for _, opIntent := range expiredOpIntents {
```
[the remark this answers is not in the document]
Comment (mickmis) [reply cmds/db-manager/cleanup/scd-evict.go:58]: I've added a section in the README, with the suggestion of implementing some batching in the future if that becomes too much of an issue.

File: cmds/db-manager/cleanup/README.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+```
+List and evict SCD expired entities
+
+Usage:
+  db-manager scd-evict [flags]
+
+Flags:
+      --delete         set this flag to true to delete the expired entities
+  -h, --help           help for scd-evict
+      --op_intents     set this flag to true to list expired operational intents (default true)
```
[the remark this answers is not in the document]
Comment (mickmis) [reply cmds/db-manager/cleanup/README.md:24]: Will do and also rename the CLI tool to just `evict` for consistency.

**Pull request merged**